### ISSUE TITLE
Wrong occurrence highlights

### ISSUE THREAD

**Issue by alexdima** [microsoft/vscode#127675]
Testing https://github.com/microsoft/vscode/issues/127362

* set up jupyter in a dev container:
  * create a new folder
  * (install remote dev container extension)
  * run `Remote Containers: Add Development Container Configuration Files...`
  * pick `Show All Definitions...` at the bottom
  * pick `Jupyter Data Science Notebooks`
  * reopen the folder in container
* follow all the Setup steps from https://github.com/microsoft/vscode/issues/127362
* find a sample notebook e.g. https://raw.githubusercontent.com/jrjohansson/scientific-python-lectures/master/Lecture-4-Matplotlib.ipynb
* execute some of the cells, enter debugging mode and set a breakpoint in a cell
* observe that clicking on a variable is highlighting random characters in the editor
* this is most likely occurrence highlighting being incorrect:

https://user-images.githubusercontent.com/5047891/123931830-2b9b1080-d991-11eb-8648-563de570532b.mp4

**Comment by roblourens** [comment]
This is from the language server right @rchiodo?

**Comment by rchiodo** [comment]
Not sure what does hover matching. It might be.

**Comment by alexdima** [comment]
To clarify, this is about clicking at a position, so we are calling a `DocumentHighlightProvider` which gives these ranges.

**Comment by rchiodo** [comment]
Yeah looking at pylance, I believe it does provide this. This would be a pylance/python extension issue then.
